assets/src/scripts/url-params.spec.js

```diff
@@ -0,0 +1,178 @@
+    });
+
+    describe('renderTimeSeriesUrlParams', () => {
+        const TEST_STATE = {
+            series: {
+                variables: {
+                    '123456': {
+                        oid: '123456',
+                        variableCode: {
+                            value: '00010'
+                        }
+                    },
+                    '123457': {
+                        oid: '123457',
+                        variableCode: {
+                            value: '00065'
+                        }
+                    }
+                },
+                methods: {
+                    '69928': {
+                        methodDescription: 'Method 69928',
+                        methodID: 69928
+                    },
+                    '69929': {
+                        methodDescription: 'Method 69929',
+                        methodID: 69929
+                    }
+                },
+                timeSeries: {
+                    '69928:current:P7D': {
+                        tsKey: 'current:P7D',
+                        method: 69928,
+                        variable: '123456'
+                    }
+                },
+                ianaTimeZone: 'America/New_York'
+            },
+            timeSeriesState: {
+                currentVariableID: '123456',
+                currentDateRange: 'P7D',
+                customTimeRange: null,
+                currentMethodID: '69928',
+                showSeries: {
+                    compare: false
+                }
+            }
+        };
+        it('adds nothing to the window.location.hash when the store is empty', () => {
+            let store = configureStore({});
+            renderTimeSeriesUrlParams(store);
+            expect(window.location.hash).toEqual('');
+        });
+
+        it('adds parameter code if current variable is set in store', () => {
+            let store = configureStore(TEST_STATE);
+            renderTimeSeriesUrlParams(store);
+            expect(window.location.hash).toContain('parameterCode=00010');
+            expect(window.location.hash).not.toContain('compare=true');
+            expect(window.location.hash).not.toContain('period');
+            expect(window.location.hash).not.toContain('startDT');
+            expect(window.location.hash).not.toContain('endDT');
+            expect(window.location.hash).not.toContain('timeSeriesId');
+        });
+
+        it('adds compare if compare is in the store', () => {
+            let store = configureStore({
+                ...TEST_STATE,
+                timeSeriesState: {
+                    ...TEST_STATE.timeSeriesState,
+                    showSeries: {
+                        compare: true
+                    }
+                }
+            });
+            renderTimeSeriesUrlParams(store);
+
+            expect(window.location.hash).toContain('parameterCode=00010');
+            expect(window.location.hash).toContain('compare=true');
```

Comment:
These comparisons between tests make it really clear what the url does/does not have.